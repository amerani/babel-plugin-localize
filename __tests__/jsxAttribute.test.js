const babel = require('@babel/core');
const plugin = require('../src/index');

const example = `
export default () => (
    <>
        <p>hello world</p>
        <Title id="titleId" label="lorem ipsum" />
    </>
);
`;

test('should transform attribute', () => {
  const { code } = babel.transform(example, {
    plugins: [[plugin, {
      elementsReplaceStringAttributes: {
        Title: ['label'],
      },
    }]],
  });
  expect(code).toMatchSnapshot();
});
