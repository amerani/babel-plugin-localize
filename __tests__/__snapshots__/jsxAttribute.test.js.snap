// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoud transform attribute 1`] = `
"export default (() => <>
        <p>{localize(\\"0\\")}</p>
        <Title id=\\"titleId\\" label={localize(\\"1\\")} />
    </>);
export const localizeKeyMap = {
  0: \\"hello world\\",
  1: \\"lorem ipsum\\"
};"
`;

exports[`should transform attribute 1`] = `
"export default (() => <>
        <p>{localize(\\"0\\")}</p>
        <Title id=\\"titleId\\" label={localize(\\"1\\")} />
    </>);
export const localizeKeyMap = {
  0: \\"hello world\\",
  1: \\"lorem ipsum\\"
};"
`;
